apps/web/app/_hooks/useSeatReservation.ts

```diff
@@ -0,0 +1,76 @@
+import { type ReservationRequestBody, type ReservedResponse } from "@repo/types";
+import { type UseMutateFunction, useMutation, useQueryClient } from "@tanstack/react-query";
+import { createSeatReservationData, deleteReservationData, modifyReservationData } from "@/api/reservation";
+import { notify } from "../store/useToastStore";
+
+interface SeatReservationMutations {
+  createSeatReservation: UseMutateFunction<
+    ReservedResponse,
+    Error,
+    { seatId: string; reservationData: ReservationRequestBody }
+  >;
+  deleteSeatReservation: UseMutateFunction<unknown, Error, string | null>;
+  modifySeatReservation: UseMutateFunction<
+    { deleteResult: ReservedResponse; createResult: ReservedResponse },
+    Error,
+    {
+      seatId?: string;
+      reservationData?: ReservationRequestBody;
+      reservationId?: string | null;
+    }
+  >;
+}
+
+export const useSeatReservation = (onSuccess?: () => void): SeatReservationMutations => {
+  const queryClient = useQueryClient();
+
+  const invalidateReservationQueries = async (): Promise<void> => {
+    await Promise.all([
+      queryClient.invalidateQueries({ queryKey: ["seats"] }),
+      queryClient.invalidateQueries({ queryKey: ["user", "reservations"] }),
+    ]);
+  };
+
+  // 좌석 예약 생성 뮤테이션
+  const { mutate: createSeatReservation } = useMutation({
+    mutationFn: createSeatReservationData,
+    onSuccess: (response) => {
+      void invalidateReservationQueries();
+      notify("success", response.message);
+      onSuccess?.();
+    },
+    onError: (error) => {
+      notify("error", `오류 발생: ${error.message}`);
+    },
+  });
+
+  // 좌석 예약 삭제 뮤테이션
+  const { mutate: deleteSeatReservation } = useMutation({
+    mutationFn: deleteReservationData,
+    onSuccess: () => {
+      void invalidateReservationQueries();
+      notify("success", "자리 예약을 삭제했습니다");
+    },
+    onError: (error) => {
+      notify("error", `오류 발생: ${error.message}`);
+    },
+  });
+
+  // 좌석 예약 수정 뮤테이션
+  const { mutate: modifySeatReservation } = useMutation({
+    mutationFn: modifyReservationData,
+    onSuccess: () => {
+      void invalidateReservationQueries();
+      notify("success", "좌석 예약 성공!");
+    },
+    onError: (error) => {
+      notify("error", `오류 발생: ${error.message}`);
+    },
+  });
```

Comment:
```suggestion
  const { mutate: deleteSeatReservation } = useMutation({
    mutationFn: deleteReservationData,
    onSuccess: (response) => {
      void invalidateReservationQueries();
      notify("success", response.message);
      onSuccess?.();
    },
    onError: (error) => {
      notify("error", `오류 발생: ${error.message}`);
    },
  });

  // 좌석 예약 수정 뮤테이션
  const { mutate: modifySeatReservation } = useMutation({
    mutationFn: modifyReservationData,
    onSuccess: (response) => {
      void invalidateReservationQueries();
      notify("success", response.message);
      onSuccess?.();
    },
    onError: (error) => {
      notify("error", `오류 발생: ${error.message}`);
    },
  });
```

사소하지만 createSeatReservation에 맞춰서 서버에서 던져주는 메세지랑 옵셔널로 받는 onSuccess를 설정해줘도 괜찮을거같고...

Reply:
옵셔널로 받는 onSuccess는 왜 해야되는거야??

Reply:
params로 받게 함수를 만들어놨는데 사용하고 있지 않는거같아서

Reply:
오 조금 더 쉽게 설명해줄 수 잇겟니?